README.md

```diff
@@ -120,131 +168,156 @@ Pre-requisites
 ```
-$ npx cordova emulate ios
-AND/OR
-$ npx cordova emulate android
+npm run <type_of_build>
 ```
 
-Creating logos
----
-If you are building your own version of the app, you must have your own logo to
-avoid app store conficts. Updating the logo is very simple using the [`ionic
-cordova resources`](https://ionicframework.com/docs/v3/cli/cordova/resources/)
-command.
+For instance: (build-dev-android)
+```
+npm run build-dev-android
+```
+Your output should look something like this:
+![Build Successful Message screenshot](Build_ss.png)
 
-**Note**: You may have to install the [`cordova-res` package](https://github.com/ionic-team/cordova-res) for the command to work
+## 4. End to End Testing
 
+A lot of the visualizations that we display in the phone client come from the server. In order to do end to end testing, we need to run a local server and connect to it. Instructions for:
 
-Troubleshooting
----
-- Make sure to use `npx ionic` and `npx cordova`. This is
-  because the setup script installs all the modules locally in a self-contained
-  environment using `npm install` and not `npm install -g`
-- Check the CI to see whether there is a known issue
-- Run the commands from the script one by one and see which fails
-    - compare the failed command with the CI logs
-- Another workaround is to delete the local environment and recreate it
-    - javascript errors: `rm -rf node_modules && npm install`
-    - native code compile errors: `rm -rf plugins && rm -rf platforms && npx cordova prepare`
+1. installing a local server,
+2. running it, 
+3. loading it with test data, and
+4. running analysis on it
 
-Beta-testing debugging
----
+are available in the [e-mission-server README](https://github.com/e-mission/e-mission-server/blob/master/README.md).
+
+In order to make end to end testing easy, if the local server is started on a HTTP (versus HTTPS port), it is in development mode.  By default, the phone app connects to the local server (localhost on iOS, [10.0.2.2 on android](https://stackoverflow.com/questions/5806220/how-to-connect-to-my-http-localhost-web-server-from-android-emulator-in-eclips)) with the `prompted-auth` authentication method. To connect to a different server, or to use a different authentication method, you need to create a `www/json/connectionConfig.json` file. More details on configuring authentication [can be found in the docs](https://github.com/e-mission/e-mission-docs/blob/master/docs/install/configuring_authentication.md).
+
+One advantage of using `skip` authentication in development mode is that any user email can be entered without a password. Developers can use one of the emails that they loaded test data for in step (3) above. So if the test data loaded was with `-u shankari@eecs.berkeley.edu`, then the login email for the phone app would also be `shankari@eecs.berkeley.edu`.
```

Comment:
This is also now different given the one-step opcode. Have you actually successfully run end-to-end with these instructions?